### PR TITLE
Updating maintainers list

### PR DESCRIPTION
**Pull request by rbhavna**
### Description
Updating maintainers list to add [samuel-oci](https://github.com/samuel-oci/)
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Zhangxunmt — APPROVED** [review]

File: MAINTAINERS.md [hunk 1]
```diff
@@ -5,18 +5,19 @@ This document contains a list of maintainers in this repo. See [opensearch-proje
...
+| Dhrubo Saha          | [dhrubo-os](https://github.com/dhrubo-os)           | Amazon    |
+| Jing Zhang           | [jngz-es](https://github.com/jngz-es)               | Amazon    |
+| Junshen Wu           | [wujunshen](https://github.com/wujunshen)           | Amazon    |
+| Sicheng Song         | [b4sjoo](https://github.com/b4sjoo)                 | Amazon    |
+| Xun Zhang            | [Zhangxunmt](https://github.com/Zhangxunmt)         | Amazon    |
+| Yaliang Wu           | [ylwu-amzn](https://github.com/ylwu-amzn)           | Amazon    |
+| Zan Niu              | [zane-neo](https://github.com/zane-neo)             | Amazon    |
+| Austin Lee           | [austintlee](https://github.com/austintlee)         | Aryn      |
+| Henry Lindeman       | [HenryL27](https://github.com/HenryL27)             | Aryn      |
+| Samuel Herman        | [samuel-oci](https://github.com/samuel-oci/)        | Oracle          |
```
Comment (ylwu-amzn) [inline MAINTAINERS.md:20]: @samuel-oci help review if these information is correct ?

**Review by samuel-oci — APPROVED** [review]

File: MAINTAINERS.md (hunk 1 above)
Comment (samuel-oci) [reply MAINTAINERS.md:20]: lgtm

**Review by ylwu-amzn — APPROVED** [review]

**Comment by dhrubo-os** [comment]
don't we need to update here: https://github.com/opensearch-project/ml-commons/blob/main/.github/CODEOWNERS ?

**Comment by ylwu-amzn** [comment]
> don't we need to update here: https://github.com/opensearch-project/ml-commons/blob/main/.github/CODEOWNERS ?

Good catch, send out another PR https://github.com/opensearch-project/ml-commons/pull/1951/files

File: MAINTAINERS.md (hunk 1 above)
Comment (peterzhuamazon) [inline MAINTAINERS.md:20]: Please fix spacing.

**Pull request merged**